lib/po_libc_wrappers.c

```diff
@@ -107,7 +108,11 @@ open(const char *path, int flags, ...)
 	mode = va_arg(args, int);
 	rel = find_relative(path, NULL);
 
-	return openat(rel.dirfd, rel.relative_path, flags, mode);
+	// If the file is already opened, no need of relative opening!
+	if( strcmp(rel.relative_path,".") != 0 )
```

Comment:
Given the comment, I'd prefer the `if` statement's condition to be the other way around (i.e., "if the path is `.`, do something special, otherwise do the `openat` thing).
 
Secondly, in this file we use defensive braces around single-statement clauses. This is now allowed by [http://nixdoc.net/man-pages/FreeBSD/man9/style.9.html](style(9)) as long as it's consistent with the surrounding code (which it would be, in this case).

Thirdly, there should be a space between `if` and `(` but no space between `(` and `strcmp`.